src/app/GitUI/LeftPanel/CachedFavorites.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.Text.Json;
+
+namespace GitUI.LeftPanel
+{
+    public class CachedFavorites
+    {
+        private readonly HashSet<string> _favorites = new();
+        private readonly object _lock = new();
+        private bool _isLoaded;
+
+        public IReadOnlyCollection<string> Favorites => _favorites;
+
+        public string Location { get; set; } = string.Empty;
+
+        private string ConfigFile
+        {
+            get => Path.Combine(Location, "favorites");
+        }
+
+        public void Add(string branch)
+        {
+            if (string.IsNullOrEmpty(branch))
+            {
+                return;
+            }
+
+            lock (_lock)
+            {
+                if (_favorites.Add(branch))
+                {
+                    Save();
+                }
+            }
+        }
+
+        public void Remove(string branch)
+        {
+            if (string.IsNullOrEmpty(branch))
+            {
+                return;
+            }
+
+            lock (_lock)
+            {
+                if (_favorites.Remove(branch))
+                {
+                    Save();
+                }
+            }
+        }
+
+        public bool Contains(string branch)
+        {
+            if (string.IsNullOrEmpty(branch))
+            {
+                return false;
+            }
+
+            LoadIfNeeded();
+
+            lock (_lock)
+            {
+                return _favorites.Contains(branch);
+            }
+        }
+
+        private void Load()
+        {
+            if (_isLoaded || !File.Exists(ConfigFile))
```

Comment:
`_isLoaded` is checked `LoadIfNeeded`. `Load` should do it unconditionally.

```suggestion
            if (!File.Exists(ConfigFile))
```